src/util-base64.c

```diff
@@ -156,11 +156,26 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
         }
     }
 
-    if (!valid && mode == BASE64_MODE_RFC4648) {
-        padding = B64_BLOCK - bbidx;
-        *decoded_bytes += ASCII_BLOCK - padding;
+    if (bbidx > 0 && bbidx < 4 && ((!valid && mode == BASE64_MODE_RFC4648))) {
+        /* Decoded bytes for 1 or 2 base64 encoded bytes is 1 */
+        padding = bbidx > 1 ? B64_BLOCK - bbidx : 2;
+        uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
+        if (dest_size < *decoded_bytes + numDecoded_blk) {
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            return ecode;
+        }
+        /* if the destination size is not at least 3 Bytes long, it'll give a dynamic
+         * buffer overflow while decoding, so, return and let the caller take care of the
+         * remaining bytes to be decoded which should always be < 4 at this stage */
+        if (dest_size - *decoded_bytes < 3)
+            return BASE64_ECODE_OK;
```

Comment:
is this not an error?

Reply:
My idea here was that if 2 bytes are permissible in dest_size, we return to the caller and let it take care of those two bytes in the next iteration/buffer like we do with the remaining bytes in MIME decoder. But, I certainly missed that this part is not called by MIME at all and the base64 decode keyword has no buffering mechanism.
So, yeah, this should probably be an error indeed.